### PR TITLE
refactor(tree): more usable component

### PR DESCRIPTION
**Pull request by anxobotana**
Adds ability to:
- style the tree item with the prop (showing defaults and options): 
```
itemStyle = {
        spacingY: 'none',      ('none' | 'x-small' | 'small' | 'medium' | 'large' | 'x-large')
        contentHight: 'single-line',      ('content-fit' | 'single-line')
        shadow: 'none',      ('none' | 'x-small' | 'small' | 'medium' | 'large' | 'x-large')
        borderRadius: 'small',      ('none' | 'x-small' | 'small' | 'medium' | 'large' | 'x-large')
        borderWidth: 'none',      ('none' | 'x-small' | 'small' | 'medium' | 'large')
        borderStyle: 'none',       ('solid' | 'dashed' | 'dotted' | 'none')
       activeColorStyle: 'neutral', ('neutral' | 'soft' | 'none')
},
showContentWhileDragging: false      (true | false)
dragHandlerPosition: 'left'             ('left' | 'right' | 'none')
showDragHandlerOnHoverOnly: true.  (true | false)
```
- `dragHandlerPosition`: drag handler can be placed on the `left`, on the `right` or `none` if we want to drag anywhere within item (defaults to `left`)
- `showDragHandlerOnHoverOnly`: drag handler can be shown always or on hover (defaults to `on hover`)
- `showContentWhileDragging`: indicates we want to see the component in the dragged position while moving it around. (defaults `false`)

By default it stays as it was ensuring backwards compatibility.

### CODE REVIEW

File: src/components/Tree/TreeItem/TreeItem.tsx [hunk 1]
```diff
@@ -319,15 +328,32 @@ export const TreeItem = memo(
...
         };
 
-        const backgroundStyle = {
-            marginLeft: -1 * (depthPadding ?? levelPadding),
-        };
+        const backgroundStyle = itemStyle?.containerClassNames
+            ? {}
+            : {
+                  marginLeft: -1 * (depthPadding ?? levelPadding),
+              };
```
Comment (SamuelAlev) [inline src/components/Tree/TreeItem/TreeItem.tsx:335]:
```suggestion
            : { marginLeft: -1 * (depthPadding ?? levelPadding) };
```

File: src/components/Tree/TreeItem/TreeItem.tsx [hunk 2]
```diff
@@ -373,6 +391,8 @@ export const TreeItem = memo(
                     )}
 
                     {showContent && contentComponent}
+
+                    {dragHandlerPosition === 'RIGHT' && dragHandler}
```
Comment (SamuelAlev) [inline src/components/Tree/TreeItem/TreeItem.tsx:395]:
```suggestion
                    {dragHandlerPosition === 'right' && dragHandler}
```

File: src/components/Tree/TreeItem/TreeItem.tsx [hunk 3]
```diff
@@ -73,6 +73,12 @@ export const TreeItem = memo(
         registerNodeChildren,
         unregisterNodeChildren,
         draggable: itemDraggable = true,
+        showDragHandlerOnHoverOnly = true,
+        dragHandlerPosition = 'LEFT',
```
Comment (SamuelAlev) [inline src/components/Tree/TreeItem/TreeItem.tsx:77]:
```suggestion
        dragHandlerPosition = 'left',
```
Let's relax and not scream

File: src/components/Tree/TreeItem/TreeItem.tsx [hunk 4]
```diff
@@ -271,18 +277,20 @@ export const TreeItem = memo(
             return {
                 liClassName: merge([
                     FOCUS_VISIBLE_STYLE,
-                    'tw-relative tw-cursor-default tw-transition-colors tw-outline-none tw-ring-inset tw-group tw-px-2.5 tw-no-underline tw-leading-5 tw-h-10',
+                    'tw-box-content tw-relative tw-cursor-default tw-transition-colors tw-outline-none tw-ring-inset tw-group tw-px-2.5 tw-no-underline tw-leading-5 tw-h-10',
                     !isActive && isSelected ? 'tw-font-medium tw-text-box-neutral-strong-inverse' : 'tw-text-text',
+                    itemStyle?.spacingY ? `tw-my-${itemStyle?.spacingY}` : 'tw-my-0',
```
Comment (SamuelAlev) [inline src/components/Tree/TreeItem/TreeItem.tsx:282]: Won't work https://tailwindcss.com/docs/content-configuration#dynamic-class-names

File: src/components/Tree/types.ts [hunk 5]
```diff
@@ -55,6 +61,12 @@ type TreeItemBaseProps = {
     accepts?: string;
     children?: ReactNode;
     draggable?: boolean;
+    showDragHandlerOnHoverOnly?: boolean;
+    dragHandlerPosition?: 'LEFT' | 'RIGHT';
+    itemStyle?: {
+        spacingY?: number | string; // tw-my-# value: 'px', 0, 1, 2, ....
+        containerClassNames?: string; // Classes to add to the item container, for example: tw-rounded-md tw-border-2 tw-border-black tw-shadow
```
Comment (SamuelAlev) [inline src/components/Tree/types.ts:68]: Not allowed to pass classnames

**Review by Kenny806 — APPROVED** [review]
Tested the current usage in Analytics Dashboard and works as expected :+1:

EDIT: I approved by mistake, but can't figure out how to dismiss the approval. Sorry about that.

**Review by fabriciomendonca — APPROVED** [review]
lg2m

**Review by imoutaharik — APPROVED** [review]

**Pull request merged**